mabwiser/approximate.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
```

Comment:
I have it separate so that if we add other ANNs the neighbors.py won't become unmanageably long. I'll see what can be refactored, the trouble is a lot of the algorithm-specific logic is nested inside the duplicate code.

Reply:
OR, from a devil's advocate perspective, one might suggest don't worry about the abstraction until unless we are adding the "second" approximate neighbors. If we never need that (which is possible..), we skip the effort to code/maintain the abstract class in the first place.